### ISSUE TITLE
feat(datepicker): Change dateformat when changing locale datepicker

### ISSUE THREAD

**Issue by rjwijnen** [valor-software/ngx-bootstrap#2806]
It would be nice when the format of the date is immediately changed when the locale/config is changed. Now you need to select a (new) date from the calendar before the format in the input field is changed to the corresponding format.

Use case:
When using a translate service you could have a default language and the language you get from an external service. For example you have default lang: 'en' and based on a User service you get the correct language for the user. The datepicker will always use the default instead of the lang from the user service in the first run. From a user perspective you want that as soon as the language is changed the format of the date is changed to and not after you select a date. I think you should update everything as soon as something for the config changes.

**Comment by sonukapoor** [comment]
I am taking a look at this issue.

**Comment by sonukapoor** [comment]
@rjwijnen Isn't that already supported using this demo?

https://valor-software.com/ngx-bootstrap/#/datepicker#locales

**Comment by rjwijnen** [comment]
@sonukapoor No it isn't. The dateformat only changes after selecting a (new) date instead of changing immediatelly when changing the locale.

**Comment by sonukapoor** [comment]
I see the calendar changes when I switch to a different language, but the date never changes for me actually even after selecting a different date. I will take a look at this.

**Comment by rjwijnen** [comment]
@sonukapoor the date format changes after selecting in the new beta version. So the only thing that does not work is the immediate change of the format when you change the locale. I thought this had something to do with the moment when the changed (bs)config is used.

**Comment by sonukapoor** [comment]
No worries, I will check it out.

**Comment by valorkin** [comment]
This is so tricky

**Comment by will421** [comment]
I have an issue (I think) very closely related, I can open a new one if necessary.

I'm on version 2.0.0-beta.8.
As @rjwijnen said, the format change after selecting a new date when locale change.
In my case, the format is not set for the initial value then set when selecting.

Here is a stackblitz reproducing this : https://stackblitz.com/edit/ngx-bootstrap-datepicker-tpmm9q
I set in app.module the local to 'fr'. But the initial value seems to be format 'en' and it switch to 'fr' when selecting a new value.

**Comment by sonukapoor** [comment]
@will421 This seems to be a different issue. Can you please reopen a new issue?

**Comment by ppossanzini** [comment]
I've exactly the same issue

**Comment by valorkin** [comment]
there are several similar PRs to fix this issue, so fix should be published this week

**Comment by rjwijnen** [comment]
It looks like this is working properly with the new localeService. Thanks!

**Comment by valorkin** [comment]
@rjwijnen yes, it's. thanks for closing, it's hard to follow all duplicated or similar issues :(
